engine/table/src/main/java/io/deephaven/engine/table/impl/MultiJoinTableImpl.java

```diff
@@ -489,33 +545,39 @@ protected void process() {
                 boolean rowModified = false;
                 int numberOfOriginalNulls = 0;
                 Arrays.fill(modifiedTablesOnThisRow, false);
-                for (int ii = 0; ii < currentRedirections.length; ++ii) {
-                    if (currentRedirections[ii] != NULL_ROW_KEY) {
+                for (int tableNumber = 0; tableNumber < tableCount; ++tableNumber) {
+                    if (currentRedirections[tableNumber] != NULL_ROW_KEY) {
                         allNull = false;
                     }
-                    if (originalValues[ii] == NULL_ROW_KEY) {
+                    if (previousRedirections[tableNumber] == NULL_ROW_KEY) {
                         numberOfOriginalNulls++;
                     }
-                    if (originalValues[ii] == MultiJoinModifiedSlotTracker.SENTINEL_UNINITIALIZED_KEY) {
-                        if (currentRedirections[ii] == NULL_ROW_KEY) {
+                    if (previousRedirections[tableNumber] == MultiJoinModifiedSlotTracker.SENTINEL_UNINITIALIZED_KEY) {
+                        if (currentRedirections[tableNumber] == NULL_ROW_KEY) {
+                            // This slot was previously deleted and may need to be reincarnated. This redirection
+                            // previously and currently points to no row and should be considered previously null
+                            // for this purpose.
                             numberOfOriginalNulls++;
+                        } else {
+                            rowModified |= (flagValues[tableNumber] & notShift) != 0;
                         }
-                        rowModified |= (flagValues[ii] & notShift) != 0;
                     } else {
                         // If the redirection has changed and we have done anything other than a shift, we must light
                         // up all the columns for the table as a modification. Similarly, if the row was added and
                         // deleted from the original table, then we must also light up all the columns as modified.
-                        if ((flagValues[ii] & addOrRemove) != 0 || currentRedirections[ii] != originalValues[ii]) {
-                            rowModified |= (modifiedTablesOnThisRow[ii] = (flagValues[ii] & notShift) != 0);
+                        if ((flagValues[tableNumber] & addOrRemove) != 0
```

Comment:
I'm curious as to why the proposed modification didn't work.

Reply:
I re-implemented and retested and it does work correctly.  I inadvertently reverted this trying to resolve another error.